src/m365/spo/commands/file/file-list.ts

```diff
@@ -70,46 +81,113 @@ class SpoFileListCommand extends SpoCommand {
     }
 
     try {
-      const files = await this.getFiles(args.options.folder, args);
-      logger.log(files.value);
+      // If --recursive option is specified, retrieve both Files and Folder details, otherwise only Files.
+      const subfolderFiles: FileProperties[] = [];
+      if (args.options.recursive) {
+        const subFolders = await this.getSubfoldersPages(args.options.folder, args, 0);
+
+        for (const folder of subFolders) {
```

Comment:
Let's compile an array of all the paths that we are going to execute `getFiles` on. The array will contain the rootfolder and the subfolders. 

Doing it this way we may use just one place to execute `getFiles`.

Reply:
That's a great idea! 💡